source/gameengine/Expressions/intern/ListValue.cpp

```diff
@@ -598,6 +600,54 @@ PyObject *CListValue::Pyget(PyObject *args)
 	return def;
 }
 
+PyObject *CListValue::PygetMatching(PyObject *args)
+{
+	const char *namestr = "";
+	const char *propstr = "";
+
+	if (!PyArg_ParseTuple(args, "s|s:getMatching", &namestr, &propstr)) {
+		return nullptr;
+	}
+
+	if (strlen(namestr) == 0 && strlen(propstr) == 0) {
+		PyErr_SetString(PyExc_ValueError, "CList.getMatching(name, prop): empty expressions.");
+		return nullptr;
+	}
+
+	std::regex namereg;
+	std::regex propreg;
+	try {
+		namereg = std::regex(namestr);
+		propreg = std::regex(propstr);
+	}
+	catch (const std::regex_error& error) {
+		PyErr_Format(PyExc_ValueError, "CList.getMatching(name, prop): invalid expression: %s.", error.what());
+		return nullptr;
+	}
+
+	CListValue *result = new CListValue();
+	result->SetReleaseOnDestruct(false);
+
+	for (CValue *item : m_pValueArray) {
+		if (strlen(namestr) == 0 || std::regex_match(item->GetName(), namereg)) {
+			if (strlen(propstr) == 0) {
+				result->Add(item);
+			}
+			else {
+				const std::vector<std::string> propnames = item->GetPropertyNames();
+				for (const std::string& propname : propnames) {
+					if (std::regex_match(propname, propreg)) {
+						result->Add(item);
+						break;
```

Comment:
Can't we do something like that:
if (result->find(item)) { continue; } so the item is not added several times but other items can be added to result list?